webapp/src/components/dot_menu/dropdown.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Strategy} from '@floating-ui/react';
+import {
+    autoUpdate,
+    flip,
+    FloatingFocusManager,
+    FloatingPortal,
+    offset,
+    Placement,
+    shift,
+    useDismiss,
+    useFloating,
+    useInteractions,
+    useRole,
+} from '@floating-ui/react-dom-interactions';
+import React, {cloneElement, ComponentProps, useState} from 'react';
+import styled from 'styled-components';
+
+const FloatingContainer = styled.div`
+    min-width: 16rem;
+    z-index: 50;
+`;
+
+type DropdownProps = {
+    target: JSX.Element;
+    children: React.ReactNode;
+    placement?: Placement;
+    offset?: Parameters<typeof offset>[0];
+    flip?: Parameters<typeof flip>[0];
+    shift?: Parameters<typeof shift>[0];
+    focusManager?: boolean | Omit<ComponentProps<typeof FloatingFocusManager>, 'context' | 'children'>;
+    portal?: boolean;
+    strategy?: Strategy;
+    isOpen: boolean;
+    onOpenChange?: ((open: boolean) => void);
+};
+
+const Dropdown = (props: DropdownProps) => {
```

Comment:
Looks like a lot of work for a "mere" dropdown :(

Reply:
Yep. I've removed some of what we don't need, but I kept some that we aren't using right now in anticipation that we'll need it later.